<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

# Laboratorio Pandas (EDA y Unión de Datos)

## Los datos

Para este laboratorio, trabajarás con los siguientes conjuntos de datos:

- **`netflix_originals.csv`**: Contiene información sobre producciones originales de Netflix. Las columnas que nos encontraremos en este conjunto de datos son: 

   - `Title`: Contiene el nombre del título de la producción original de Netflix.

   - `Genre`: Especifica el género o combinación de géneros de la producción.

   - `Premiere`: Indica la fecha de estreno de la producción en Netflix.

   - `Runtime`: Contiene la duración de la producción en minutos.

   - `IMDB Score`: Representa la calificación promedio que la producción ha recibido en la plataforma IMDB.

   - `Language`: Especifica el idioma principal o los idiomas en los que se presenta la producción.

- **`netflix_titles.csv`**: Contiene información sobre una variedad de títulos disponibles en la plataforma Netflix, incluyendo tanto películas como series de televisión. Las columnas que nos encontraremos en este conjunto de datos son:

   - `show_id`: Identificador único para cada título en el dataset.

   - `type`: Especifica el tipo de contenido, ya sea "Movie" (película) o "TV Show" (serie de televisión).

   - `title`: Contiene el nombre del título de la producción.

   - `director`: Muestra el nombre del director del título. Puede estar vacío en algunos casos.

   - `cast`: Lista de actores y actrices que participaron en el título. También puede estar vacío en algunos casos.

   - `country`: País o países donde se produjo el título. Esta columna también puede tener valores faltantes.

   - `date_added`: Fecha en la que el título fue añadido al catálogo de Netflix.

   - `release_year`: Año en que el título fue lanzado o producido.

   - `rating`: Clasificación del contenido en términos de audiencia (por ejemplo, PG-13, TV-MA).

   - `duration`: Duración del título. Para películas, se expresa en minutos (e.g., "90 min") y para series en temporadas (e.g., "2 Seasons").

   - `listed_in`: Categorías o géneros bajo los cuales el título está clasificado (por ejemplo, "Documentaries", "TV Dramas").

   - `description`: Breve sinopsis o descripción del contenido del título.

A lo largo de los laboratorios de Pandas, trabajarás con estos conjuntos de datos (o con los derivados del Laboratorio), algunas de las actividades que realizaremos son:

- Exploración de datos y análisis exploratorio (EDA) para familiarizarse con los conjuntos de datos.

- Realización de uniones entre conjuntos de datos, aplicando diferentes técnicas de *merging* y analizando la relevancia de cada método.

- Uso de funciones de selección y filtrado (`loc`, `iloc`) para extraer información clave.

- Desarrollo de análisis descriptivos y visualización de patrones para comprender mejor las decisiones estratégicas de la plataforma.

----

## Objetivo General de la Serie de Laboratorios



1. **Unión de `netflix_titles.csv` con `netflix_originals.csv`:**

   - **Análisis de contenido original:**  Identificarás los títulos que son producciones originales de Netflix y los compararás con los títulos adquiridos. Este análisis te permitirá:

     - Comparar la diversidad de géneros entre los títulos originales y no originales.

     - Evaluar si ciertos géneros o tipos de contenido (como documentales o series) son más frecuentes en las producciones originales.

     - Investigar si la producción original se concentra en determinados países o idiomas.


   - **Evaluación de contenido:** En esta etapa, explorarás si las producciones originales de Netflix tienden a recibir mejores o peores evaluaciones en comparación con el contenido adquirido. Algunas preguntas que se buscarán responder incluyen:

     - ¿Los títulos originales tienen una calificación promedio superior en plataformas como IMDB?

     - ¿Son las producciones originales más consistentes en términos de calidad percibida?

     - ¿Existen diferencias en la popularidad de los títulos originales según su género o país de origen?


   - **Identificación de tendencias:**  Por último, este análisis te permitirá explorar cómo ha evolucionado la estrategia de contenido de Netflix a lo largo del tiempo. Algunos puntos clave a investigar serán:

     - ¿Cómo ha crecido la producción original en comparación con la adquisición de contenido?

     - ¿Qué géneros o tipos de contenido han recibido mayor inversión en los últimos años?
     
     - ¿Existen patrones en los países o idiomas de las producciones originales, especialmente en relación con la expansión global de Netflix?


----


## Ejercicio: Análisis y Enriquecimiento del Catálogo de Netflix

### Parte 1: Análisis Exploratorio de Datos (EDA)

1. **Carga de los datos:**

   - Deberás cargar los dos conjuntos de datos que te hemos proporcionado para realizar este Laboratorio:

     - `netflix_originals.csv`

     - `netflix_titles.csv`

   Deberás revisar la estructura de cada conjunto de datos utilizando los métodos aprendidos durante la lección.

2. **Análisis de columnas y consistencia:**

   - Deberás hacer un análisis exploratorio de cada uno de los conjuntos de datos. Será obligatorio que incluyas las conclusiones generales obtenidas de cada uno de los análisis exploratorios en el Jupyter Notebook de tus soluciones.

   - Explorar qué columnas contiene cada conjunto de datos y cuáles son sus tipos de datos. Identificar si hay valores nulos o duplicados.

   - Algunas preguntas que nos podemos plantear en esta fase son (recuerda que esto es solo un ejemplo, y que podemos explorar todo lo que queramos/necesitemos para entender lo mejor posible los conjuntos de datos):

      - ¿Cuáles son los tipos de datos de cada columna? Nos puede ayudar a detectar si algún tipo de dato está incorrectamente asignado, como fechas almacenadas como cadenas de texto.

      - ¿Existen valores nulos? ¿En qué columnas están? Este paso es crucial para decidir si deben ser eliminados, rellenados o ignorados.

      - ¿Cuántos títulos únicos hay en cada conjunto de datos? Nos puede permitir verificar si existen títulos duplicados que puedan distorsionar el análisis.

      - ¿Qué géneros tenemos en el conjunto de datos? Esto nos puede ayudar a identificar los géneros más populares o incluso a identificar una posible limpieza de dicha columna para poder extraer conclusiones más claras en nuestro análisis.

### Parte 2: Unión de Datos

1. **Realización de la unión:**

   - Deberás realizar la unión que consideres más apropiada entre `netflix_titles.csv` y `netflix_originals.csv`. Además, deberás justificar el tipo de unión que has realizado.

   - Esto nos ayudará a contestar preguntas más adelente del tipo:

     - ¿Cuántos títulos en `netflix_titles.csv` son producciones originales de Netflix?

     - ¿Hay diferencias significativas en la puntuación de IMDB entre títulos originales y no originales?

### Parte 3: Uso de `loc` e `iloc`

Usando el DataFrame creado en la Parte 2: 

1. **Filtrado de datos con `loc`:**

   - Selecciona todas las filas donde el tipo de contenido sea “Movie”.

   - Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.

   - Selecciona todas las filas donde el país sea “United States”.

   - Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

   - Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

2. **Acceso a datos específicos con `iloc`:**

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.

   - Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.


**NOTA IMPORTANTE**: Antes de terminar el lab, guarda el conjunto de datos creado en la fase de unión para trabajar con el en las siguientes lecciones. 

In [165]:
import pandas as pd
import numpy as np

En primer lugar cargamos ambos archivos para que serán con los que trabajemos a lo largo de este laboratorio.

In [166]:
df1_Netflix = pd.read_csv ("datos/netflix_originals.csv")
df1_Netflix.head()

Unnamed: 0            Title                  Genre           Premiere  \
0           0  Enter the Anime            Documentary     August 5, 2019   
1           1      Dark Forces               Thriller    August 21, 2020   
2           2          The App  Science fiction/Drama  December 26, 2019   
3           3   The Open House        Horror thriller   January 19, 2018   
4           4      Kaali Khuhi                Mystery   October 30, 2020   

   Runtime  IMDB Score          Language  
0       58         2.5  English/Japanese  
1       81         2.6           Spanish  
2       79         2.6           Italian  
3       94         3.2           English  
4       90         3.4             Hindi

In [167]:
df2_Netflix = pd.read_csv ("datos/netflix_titles.csv")
df2_Netflix.head()

Unnamed: 0 show_id     type                  title         director  \
0           0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1           1      s2  TV Show          Blood & Water              NaN   
2           2      s3  TV Show              Ganglands  Julien Leclercq   
3           3      s4  TV Show  Jailbirds New Orleans              NaN   
4           4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [168]:
df1_Netflix.sample(5)

Unnamed: 0                                           Title  \
452         505  Mucho Mucho Amor: The Legend of Walter Mercado   
364         399                                   Death to 2020   
407         450                          The Night Comes for Us   
455         508                                      On My Skin   
405         448                        The Christmas Chronicles   

                                  Genre            Premiere  Runtime  \
452                         Documentary        July 8, 2020       96   
364                              Comedy   December 27, 2020       70   
407                     Action-thriller    October 19, 2018      121   
455                         Crime drama  September 12, 2018      100   
405  Christmas/Fantasy/Adventure/Comedy   November 22, 2018      104   

     IMDB Score         Language  
452         7.3  Spanish/English  
364         6.8          English  
407         7.0       Indonesian  
455         7.3          Italian  
405         7.0          English

In [169]:
#Sacamos el numero de duplicados.
numero_duplicados_1=int(df1_Netflix.duplicated().sum())
numero_duplicados_2=int(df2_Netflix.duplicated().sum())
print(numero_duplicados_1, numero_duplicados_2)

0 0


In [170]:
#Sacamos los valores unicos y su frecuencia que esperamosque saea 1 en todos puesto que no habia filas duplicadas
#  segun vimos anteriormente
df1_Netflix.value_counts()

Unnamed: 0  Title                                     Genre                  Premiere           Runtime  IMDB Score  Language    
583         David Attenborough: A Life on Our Planet  Documentary            October 4, 2020    83       9.0         English         1
561         The Irishman                              Crime drama            November 27, 2019  209      7.8         English         1
560         If Anything Happens I Love You            Animation / Short      November 20, 2020  12       7.8         English         1
555         Crip Camp: A Disability Revolution        Documentary            March 25, 2020     108      7.7         English         1
553         Beasts of No Nation                       War drama              October 16, 2015   136      7.7         English/Akan    1
                                                                                                                                    ..
5           Drive                                     Action

In [171]:
df2_Netflix.value_counts()

Unnamed: 0  show_id  type   title                   director             cast                                                                                                                                                                                                                                                  country                                                                date_added          release_year  rating  duration  listed_in                                         description                                                                                                                                                                                                                                      
8806        s8807    Movie  Zubaan                  Mozez Singh          Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanana, Manish Chaudhary, Meghna Malik, Malkeet Rauni, Anita Shabdish, Chittaranjan Tripathy                                                            

In [172]:
df1_Netflix.shape

(513, 7)

In [173]:
df2_Netflix.shape


(8807, 13)

In [174]:
#  Vemos una discrepancia entre el numero de filas que nos devuelve shape y la numeración de Unnamed 0 que a priori
# podría servir de indice no es correlativa. Unnamed 0 es columna candidata a ser borrada, pero aun es un poco pronto
# para decidir eso, por riesgo a cargarnos datos



In [175]:
#Vamos a sacar info sobre cuantos valores nulos y que tipos de datos tenemos
df1_Netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  513 non-null    int64  
 1   Title       513 non-null    object 
 2   Genre       513 non-null    object 
 3   Premiere    513 non-null    object 
 4   Runtime     513 non-null    int64  
 5   IMDB Score  513 non-null    float64
 6   Language    513 non-null    object 
dtypes: float64(1), int64(2), object(4)
memory usage: 28.2+ KB


Para nuestra alegría no hay valores nulos, este frame y los tipos de datos no chirrian con lo que a priori deben contener las
columnas. Le echaremos un ojo más adelante a la duración que está en enteros en vez de expresar hora, pero para hacer
comparaciones puede ser hasta mejor

In [176]:
df2_Netflix.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Unnamed: 0    8807 non-null   int64 
 1   show_id       8807 non-null   object
 2   type          8807 non-null   object
 3   title         8807 non-null   object
 4   director      6173 non-null   object
 5   cast          7982 non-null   object
 6   country       7976 non-null   object
 7   date_added    8797 non-null   object
 8   release_year  8807 non-null   int64 
 9   rating        8803 non-null   object
 10  duration      3994 non-null   object
 11  listed_in     8807 non-null   object
 12  description   8807 non-null   object
dtypes: int64(2), object(11)
memory usage: 894.6+ KB


Sin embargo en el otro vemos que si hay muchos valores nulos en ciertas columnas y tendremos que ver como afecta eso a nuestro analisis. También chirría que el tipo de datos del rating y la duración sean objetos en vez de float y int respectivamente

Vamos a correr el riesgo de modificar el dataframe para lo cual como media de prudencia usaremos otra variable
para perseverar el original

In [177]:
# Eliminamos la columna Unnamed 0 que no parece aportar nada a lo que nos piden en nuestro análisis
df1_modificado = df1_Netflix.drop(columns=["Unnamed: 0"])
df2_modificado = df2_Netflix.drop(columns=["Unnamed: 0"])

In [178]:
df1_modificado

Title                  Genre  \
0                                Enter the Anime            Documentary   
1                                    Dark Forces               Thriller   
2                                        The App  Science fiction/Drama   
3                                 The Open House        Horror thriller   
4                                    Kaali Khuhi                Mystery   
..                                           ...                    ...   
508                       Cuba and the Cameraman            Documentary   
509                       Dancing with the Birds            Documentary   
510  Winter on Fire: Ukraine's Fight for Freedom            Documentary   
511                      Springsteen on Broadway           One-man show   
512     David Attenborough: A Life on Our Planet            Documentary   

              Premiere  Runtime  IMDB Score                  Language  
0       August 5, 2019       58         2.5          English/Japanese  
1      August 21, 2020       81         2.6                   Spanish  
2    December 26, 2019       79         2.6                   Italian  
3     January 19, 2018       94         3.2                   English  
4     October 30, 2020       90         3.4                     Hindi  
..                 ...      ...         ...                       ...  
508  November 24, 2017      114         8.3                   English  
509   October 23, 2019       51         8.3                   English  
510    October 9, 2015       91         8.4  English/Ukranian/Russian  
511  December 16, 2018      153         8.5                   English  
512    October 4, 2020       83         9.0                   English  

[513 rows x 6 columns]

In [179]:
df2_modificado

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA        NaN   
3     September 24, 2021          2021  TV-MA        NaN   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R        NaN   
8805    January 11, 2020          2006     PG        NaN   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

In [180]:
# Vamos a ver si hay titulos duplicados
df1_modificado["Title"].unique().size

513

In [181]:
df2_modificado["title"].unique().size

8807

Luego todas las celdas hablan de peliculas diferentes

In [182]:
#Sacamos el porcentaje de películas de cada género
df1_modificado.value_counts(["Genre"], normalize=True)

Genre           
Documentary         0.257310
Drama               0.142300
Comedy              0.081871
Romantic comedy     0.068226
Thriller            0.064327
                      ...   
Superhero-Comedy    0.001949
Stop Motion         0.001949
Urban fantasy       0.001949
War-Comedy          0.001949
Zombie/Heist        0.001949
Name: proportion, Length: 106, dtype: float64

In [183]:
#En el caso del otro dataframe el genero viene recogido en listed in
df2_modificado.value_counts(["listed_in"], normalize=True)

listed_in                                               
Dramas, International Movies                                0.041104
Documentaries                                               0.040763
Stand-Up Comedy                                             0.037924
Comedies, Dramas, International Movies                      0.031112
Dramas, Independent Movies, International Movies            0.028614
                                                              ...   
TV Action & Adventure, TV Comedies                          0.000114
TV Action & Adventure, TV Comedies, TV Dramas               0.000114
Action & Adventure, Classic Movies, International Movies    0.000114
Action & Adventure, Classic Movies, Sci-Fi & Fantasy        0.000114
Action & Adventure, Comedies, Horror Movies                 0.000114
Name: proportion, Length: 514, dtype: float64

In [184]:
df1_modificado.describe()

Runtime  IMDB Score
count  513.000000  513.000000
mean    94.674464    6.210916
std     26.648121    0.968850
min      4.000000    2.500000
25%     87.000000    5.600000
50%     97.000000    6.300000
75%    108.000000    6.900000
max    209.000000    9.000000

En la tabla de arriba vemos que la duración de los originales abarca valores muy dispares, de los 4 minutos a los 209. Y vemos que la duración media, 94.6 está muy cerca de la mediana 97. Eso, nos indica una distribución relativamente cercana a la normal.

Por otro lado, observando la correlación entre duracion y calificación en IMDB vemos una tendencia a que a mayor longitud, mejor calificación en dicha palataforma de rating.

Por otro lado, la calficación media de 6.2 es relativamente alta a vida cuenta de la elevada cantidad de titulos con calificación baja.

In [185]:
df1_modificado.describe(include="O")

Title        Genre  \
count                                        513          513   
unique                                       513          106   
top     David Attenborough: A Life on Our Planet  Documentary   
freq                                           1          132   

                Premiere Language  
count                513      513  
unique               350       37  
top     October 18, 2019  English  
freq                   5      352

De esta columna destacamos que el Inglés es el idioma más comun en las producciones originales de Netflix

In [186]:
df2_modificado.describe()

release_year
count   8807.000000
mean    2014.180198
std        8.819312
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

In [187]:
df2_modificado.describe(include="O")

show_id   type   title       director                cast  \
count     8807   8807    8807           6173                7982   
unique    8807      2    8807           4528                7692   
top      s8807  Movie  Zubaan  Rajiv Chilaka  David Attenborough   
freq         1   6131       1             19                  19   

              country       date_added rating   duration  \
count            7976             8797   8803       3994   
unique            748             1767     17        185   
top     United States  January 1, 2020  TV-MA  2 Seasons   
freq             2818              109   3207        425   

                           listed_in  \
count                           8807   
unique                           514   
top     Dramas, International Movies   
freq                             362   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4

Hay que ser cautelosos al comparar ambas, porque los generos vienen recogidos demanera diferente en ambas pero se puede observar que en el archivo de originales predominan mucho más los documentales que en el otro. Una posible explicación es que a Netflix le resulta en general más barato producir documentales que películas, mientrasque en las peliculas su apuesta mayoritaria viene por unir a su catálogo peliculas que no han producidio ellos.

En todo caso, analizando la segunda tabla se ve que Netflix ofrece un catalogo de generos muy variados respondiendo probablemente a una estrategia de atraer a diferente público con diferentes gustos.

Por cuestión de tiempo no entramos en ello, pero merecería la pena analizar la duración de los títulos de ambas producciones para ver si predomina los títulos de una determinada longitud.

merge con how en outer para unir left on right para unir las dos tablas

In [188]:
df_merged=df1_modificado.merge(df2_modificado, left_on = "Title", right_on= "title", how="outer")
df_merged

Title Genre Premiere  Runtime  IMDB Score Language show_id     type  \
0      NaN   NaN      NaN      NaN         NaN      NaN   s2037    Movie   
1      NaN   NaN      NaN      NaN         NaN      NaN   s2305    Movie   
2      NaN   NaN      NaN      NaN         NaN      NaN   s2482    Movie   
3      NaN   NaN      NaN      NaN         NaN      NaN   s2325    Movie   
4      NaN   NaN      NaN      NaN         NaN      NaN   s5974    Movie   
...    ...   ...      ...      ...         ...      ...     ...      ...   
8802   NaN   NaN      NaN      NaN         NaN      NaN   s6178  TV Show   
8803   NaN   NaN      NaN      NaN         NaN      NaN   s4915  TV Show   
8804   NaN   NaN      NaN      NaN         NaN      NaN   s7102  TV Show   
8805   NaN   NaN      NaN      NaN         NaN      NaN   s5023    Movie   
8806   NaN   NaN      NaN      NaN         NaN      NaN   s7109    Movie   

                              title                     director  \
0                            #Alive                       Cho Il   
1     #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto   
2                 #FriendButMarried                Rako Prijanto   
3               #FriendButMarried 2                Rako Prijanto   
4                             #Roxy              Michael Kennedy   
...                             ...                          ...   
8802                       忍者ハットリくん                          NaN   
8803                           海的儿子                          NaN   
8804                           마녀사냥                          NaN   
8805                        반드시 잡는다                Hong-seon Kim   
8806            최강전사 미니특공대 : 영웅의 탄생                Young Jun Lee   

                                                   cast      country  \
0                              Yoo Ah-in, Park Shin-hye  South Korea   
1                           Helen Mirren, Gengher Gatti        Italy   
2     Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...    Indonesia   
3     Adipati Dolken, Mawar de Jongh, Sari Nila, Von...    Indonesia   
4     Jake Short, Sarah Fisher, Booboo Stewart, Dann...       Canada   
...                                                 ...          ...   
8802                                                NaN        Japan   
8803  Li Nanxing, Christopher Lee, Jesseca Liu, Appl...          NaN   
8804  Si-kyung Sung, Se-yoon Yoo, Dong-yup Shin, Ji-...  South Korea   
8805                                      Baek Yoon-sik  South Korea   
8806  Um Sang-hyun, Yang Jeong-hwa, Jeon Tae-yeol, S...          NaN   

              date_added  release_year rating   duration  \
0      September 8, 2020          2020  TV-MA        NaN   
1           July 1, 2020          2019  TV-14        NaN   
2           May 21, 2020          2018   TV-G        NaN   
3          June 28, 2020          2020   TV-G    104 min   
4         April 10, 2019          2018  TV-14        NaN   
...                  ...           ...    ...        ...   
8802   December 23, 2018          2012  TV-Y7  2 Seasons   
8803      April 27, 2018          2016  TV-14        NaN   
8804   February 19, 2018          2015  TV-MA        NaN   
8805   February 28, 2018          2017  TV-MA    110 min   
8806   September 1, 2018          2018  TV-Y7     68 min   

                                              listed_in  \
0        Horror Movies, International Movies, Thrillers   
1                   Documentaries, International Movies   
2         Dramas, International Movies, Romantic Movies   
3         Dramas, International Movies, Romantic Movies   
4                             Comedies, Romantic Movies   
...                                                 ...   
8802                             Anime Series, Kids' TV   
8803                  International TV Shows, TV Dramas   
8804  International TV Shows, Korean TV Shows, Stand...   
8805            Dramas, International Movies, Thrillers   
8806                           Children 

nos decantamos por un inner para la unión porque a la luz de las preguntas que vamos a contestar nos limitamos a los originales de netflix, ampliados con la información nos de la segunda tabla.

Además, asi manejamos menos cantidad de información con las ganancias consecuentes en legibilidad y tiempo de ejecución.

Por otro lado asi nos ahorramos muchos valores null presentes en las filas de los titulos no originales (segunda tabla) que restarían calidad y consistencia a nuestro analisis 

In [189]:
df_merged.describe()

Runtime  IMDB Score  release_year
count  513.000000  513.000000   8807.000000
mean    94.674464    6.210916   2014.180198
std     26.648121    0.968850      8.819312
min      4.000000    2.500000   1925.000000
25%     87.000000    5.600000   2013.000000
50%     97.000000    6.300000   2017.000000
75%    108.000000    6.900000   2019.000000
max    209.000000    9.000000   2021.000000

Por el tipo de unión de tablas que escogimos, nos quedamos con los títulos de la primera tabla que estaban incluidos en la segunda. La fila count de la tabla anterior nos dice que son 513 los titulos de Netflix que son producciones originales. Si volvemos la vista atrás recordamos que el archivo de originales tenia 513 titulos unicos lo que nos confirmaría que todos los titulos originales de Netflix están en el segundo archivo, que en ese sentido es consistente

In [190]:
# Finalmente para quitar redundancias en los datos, eliminamos las columnas title que
# ofrece datos idénticos a la primera

df_merged.drop(columns=["title","Genre","duration", "release_year"])

Title Premiere  Runtime  IMDB Score Language show_id     type  \
0      NaN      NaN      NaN         NaN      NaN   s2037    Movie   
1      NaN      NaN      NaN         NaN      NaN   s2305    Movie   
2      NaN      NaN      NaN         NaN      NaN   s2482    Movie   
3      NaN      NaN      NaN         NaN      NaN   s2325    Movie   
4      NaN      NaN      NaN         NaN      NaN   s5974    Movie   
...    ...      ...      ...         ...      ...     ...      ...   
8802   NaN      NaN      NaN         NaN      NaN   s6178  TV Show   
8803   NaN      NaN      NaN         NaN      NaN   s4915  TV Show   
8804   NaN      NaN      NaN         NaN      NaN   s7102  TV Show   
8805   NaN      NaN      NaN         NaN      NaN   s5023    Movie   
8806   NaN      NaN      NaN         NaN      NaN   s7109    Movie   

                         director  \
0                          Cho Il   
1     Sabina Fedeli, Anna Migotto   
2                   Rako Prijanto   
3                   Rako Prijanto   
4                 Michael Kennedy   
...                           ...   
8802                          NaN   
8803                          NaN   
8804                          NaN   
8805                Hong-seon Kim   
8806                Young Jun Lee   

                                                   cast      country  \
0                              Yoo Ah-in, Park Shin-hye  South Korea   
1                           Helen Mirren, Gengher Gatti        Italy   
2     Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...    Indonesia   
3     Adipati Dolken, Mawar de Jongh, Sari Nila, Von...    Indonesia   
4     Jake Short, Sarah Fisher, Booboo Stewart, Dann...       Canada   
...                                                 ...          ...   
8802                                                NaN        Japan   
8803  Li Nanxing, Christopher Lee, Jesseca Liu, Appl...          NaN   
8804  Si-kyung Sung, Se-yoon Yoo, Dong-yup Shin, Ji-...  South Korea   
8805                                      Baek Yoon-sik  South Korea   
8806  Um Sang-hyun, Yang Jeong-hwa, Jeon Tae-yeol, S...          NaN   

              date_added rating  \
0      September 8, 2020  TV-MA   
1           July 1, 2020  TV-14   
2           May 21, 2020   TV-G   
3          June 28, 2020   TV-G   
4         April 10, 2019  TV-14   
...                  ...    ...   
8802   December 23, 2018  TV-Y7   
8803      April 27, 2018  TV-14   
8804   February 19, 2018  TV-MA   
8805   February 28, 2018  TV-MA   
8806   September 1, 2018  TV-Y7   

                                              listed_in  \
0        Horror Movies, International Movies, Thrillers   
1                   Documentaries, International Movies   
2         Dramas, International Movies, Romantic Movies   
3         Dramas, International Movies, Romantic Movies   
4                             Comedies, Romantic Movies   
...                                                 ...   
8802                             Anime Series, Kids' TV   
8803                  International TV Shows, TV Dramas   
8804  International TV Shows, Korean TV Shows, Stand...   
8805            Dramas, International Movies, Thrillers   
8806                           Children & Family Movies   

                                            description  
0     As a grisly virus rampages a city, a lone man ...  
1     Through her diary, Anne Frank's story is retol...  
2     Pining for his high school crush for years, a ...  
3     As Ayu and Ditto finally transition from best ...  
4     A teenage hacker with a huge nose helps a cool...  
...                                                 ...  
8802  Hailing from the mountains of Iga, Kanzo Hatto...  
8803  Two brothers start a new life in Singapore, wh...  
8804  Four Korean celebrity men and guest stars of b...  
8805  After people in his town start turning up dead...  
8806  Miniforce, a special task force of elite range...  

[8807 rows x 14 columns]

PARTE 3

1.Filtrado de datos con loc

- Selecciona todas las filas donde el tipo de contenido sea “Movie”.

In [191]:
df_merged.loc[df_merged['type'] == 'Movie']


Title Genre Premiere  Runtime  IMDB Score Language show_id   type  \
0      NaN   NaN      NaN      NaN         NaN      NaN   s2037  Movie   
1      NaN   NaN      NaN      NaN         NaN      NaN   s2305  Movie   
2      NaN   NaN      NaN      NaN         NaN      NaN   s2482  Movie   
3      NaN   NaN      NaN      NaN         NaN      NaN   s2325  Movie   
4      NaN   NaN      NaN      NaN         NaN      NaN   s5974  Movie   
...    ...   ...      ...      ...         ...      ...     ...    ...   
8798   NaN   NaN      NaN      NaN         NaN      NaN   s5970  Movie   
8799   NaN   NaN      NaN      NaN         NaN      NaN   s5971  Movie   
8801   NaN   NaN      NaN      NaN         NaN      NaN   s4668  Movie   
8805   NaN   NaN      NaN      NaN         NaN      NaN   s5023  Movie   
8806   NaN   NaN      NaN      NaN         NaN      NaN   s7109  Movie   

                              title                     director  \
0                            #Alive                       Cho Il   
1     #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto   
2                 #FriendButMarried                Rako Prijanto   
3               #FriendButMarried 2                Rako Prijanto   
4                             #Roxy              Michael Kennedy   
...                             ...                          ...   
8798             ​Maj Rati ​​Keteki            Santwana Bardoloi   
8799                    ​Mayurakshi                  Atanu Ghosh   
8801           ​​Kuch Bheege Alfaaz                         Onir   
8805                        반드시 잡는다                Hong-seon Kim   
8806            최강전사 미니특공대 : 영웅의 탄생                Young Jun Lee   

                                                   cast      country  \
0                              Yoo Ah-in, Park Shin-hye  South Korea   
1                           Helen Mirren, Gengher Gatti        Italy   
2     Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...    Indonesia   
3     Adipati Dolken, Mawar de Jongh, Sari Nila, Von...    Indonesia   
4     Jake Short, Sarah Fisher, Booboo Stewart, Dann...       Canada   
...                                                 ...          ...   
8798  Adil Hussain, Shakil Imtiaz, Mahendra Rabha, S...        India   
8799  Soumitra Chatterjee, Prasenjit Chatterjee, Ind...        India   
8801  Geetanjali Thapa, Zain Khan Durrani, Shray Rai...        India   
8805                                      Baek Yoon-sik  South Korea   
8806  Um Sang-hyun, Yang Jeong-hwa, Jeon Tae-yeol, S...          NaN   

              date_added  release_year rating duration  \
0      September 8, 2020          2020  TV-MA      NaN   
1           July 1, 2020          2019  TV-14      NaN   
2           May 21, 2020          2018   TV-G      NaN   
3          June 28, 2020          2020   TV-G  104 min   
4         April 10, 2019          2018  TV-14      NaN   
...                  ...           ...    ...      ...   
8798  September 15, 2018          2017  TV-14  117 min   
8799  September 15, 2018          2017  TV-14      NaN   
8801   September 1, 2018          2018  TV-14  110 min   
8805   February 28, 2018          2017  TV-MA  110 min   
8806   September 1, 2018          2018  TV-Y7   68 min   

                                             listed_in  \
0       Horror Movies, International Movies, Thrillers   
1                  Documentaries, International Movies   
2        Dramas, International Movies, Romantic Movies   
3        Dramas, International Movies, Romantic Movies   
4                            Comedies, Romantic Movies   
...                                                ...   
8798                      Dramas, International Movies   
8799                      Dramas, International Movies   
8801  Dramas, Independent Movies, International Movies   
8805           Dramas, International Movies, Thrillers   
8806                          Children & Family Movies   

                                        

- Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show

In [192]:
df_merged.loc[df_merged['type'] == 'TV Show', ['title', 'duration']]

title   duration
8                                #blackAF        NaN
14                               (Un)Well        NaN
20                     100 Days My Prince        NaN
22                             100 Humans        NaN
27                            100% Hotter        NaN
...                                   ...        ...
8796                  แผนร้ายนายเจ้าเล่ห์        NaN
8800  ​SAINT SEIYA: Knights of the Zodiac  2 Seasons
8802                             忍者ハットリくん  2 Seasons
8803                                 海的儿子        NaN
8804                                 마녀사냥        NaN

[2676 rows x 2 columns]

- Selecciona todas las filas donde el país sea “United States”.

In [193]:
df_merged.loc[df_merged['country'] == 'United States']

Title Genre Premiere  Runtime  IMDB Score Language show_id     type  \
5      NaN   NaN      NaN      NaN         NaN      NaN   s5696    Movie   
8      NaN   NaN      NaN      NaN         NaN      NaN   s2667  TV Show   
10     NaN   NaN      NaN      NaN         NaN      NaN   s5278    Movie   
13     NaN   NaN      NaN      NaN         NaN      NaN   s5972    Movie   
14     NaN   NaN      NaN      NaN         NaN      NaN   s2141  TV Show   
...    ...   ...      ...      ...         ...      ...     ...      ...   
8770   NaN   NaN      NaN      NaN         NaN      NaN   s8803    Movie   
8774   NaN   NaN      NaN      NaN         NaN      NaN   s8805    Movie   
8777   NaN   NaN      NaN      NaN         NaN      NaN   s8806    Movie   
8778   NaN   NaN      NaN      NaN         NaN      NaN   s1056    Movie   
8785   NaN   NaN      NaN      NaN         NaN      NaN   s5978    Movie   

             title                                director  \
5        #Rucker50                   Robert McCullough Jr.   
8         #blackAF                                     NaN   
10    #realityhigh                        Fernando Lebrija   
13        (T)ERROR  Lyric R. Cabral, David Felix Sutcliffe   
14        (Un)Well                                     NaN   
...            ...                                     ...   
8770        Zodiac                           David Fincher   
8774    Zombieland                         Ruben Fleischer   
8777          Zoom                            Peter Hewitt   
8778     Zoot Suit                             Luis Valdez   
8785      Æon Flux                            Karyn Kusama   

                                                   cast        country  \
5                                                   NaN  United States   
8     Kenya Barris, Rashida Jones, Iman Benson, Genn...  United States   
10    Nesta Cooper, Kate Walsh, John Michael Higgins...  United States   
13                                                  NaN  United States   
14                                                  NaN  United States   
...                                                 ...            ...   
8770  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8774  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8777  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8778  Daniel Valdez, Edward James Olmos, Charles Aid...  United States   
8785  Charlize Theron, Marton Csokas, Jonny Lee Mill...  United States   

             date_added  release_year rating duration  \
5      December 1, 2016          2016  TV-PG   56 min   
8        April 17, 2020          2020  TV-MA      NaN   
10    September 8, 2017          2017  TV-14      NaN   
13        June 30, 2016          2015     NR   84 min   
14      August 12, 2020          2020  TV-MA      NaN   
...                 ...           ...    ...      ...   
8770  November 20, 2019          2007      R  158 min   
8774   November 1, 2019          2009      R      NaN   
8777   January 11, 2020          2006     PG      NaN   
8778     April 15, 2021          1981      R      NaN   
8785   February 1, 2018          2005  PG-13      NaN   

                                  listed_in  \
5              Documentaries, Sports Movies   
8                               TV Comedies   
10                                 Comedies   
13                            Documentaries   
14                               Reality TV   
...                                     ...   
8770         Cult Movies, Dramas, Thrillers   
8774                Comedies, Horror Movies   
8777     Children & Family Movies, Comedies   
8778  Cult Movies, Dramas, Music & Musicals   
8785   Action & Adventure, Sci-Fi & Fantasy   

                                            description  
5     This documentary celebrates the 50th anniversa...  
8     Kenya Barris and his family navigate relations...  
10    When nerdy high schooler Dan

 Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

   



In [194]:
df_merged.loc[df_merged['type'] == 'Movie', ['title', 'director']]

title                     director
0                            #Alive                       Cho Il
1     #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto
2                 #FriendButMarried                Rako Prijanto
3               #FriendButMarried 2                Rako Prijanto
4                             #Roxy              Michael Kennedy
...                             ...                          ...
8798             ​Maj Rati ​​Keteki            Santwana Bardoloi
8799                    ​Mayurakshi                  Atanu Ghosh
8801           ​​Kuch Bheege Alfaaz                         Onir
8805                        반드시 잡는다                Hong-seon Kim
8806            최강전사 미니특공대 : 영웅의 탄생                Young Jun Lee

[6131 rows x 2 columns]

- Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.


In [195]:
df_merged.loc[df_merged['release_year'] == 2018, ['title', 'listed_in']]


title                                          listed_in
2        #FriendButMarried      Dramas, International Movies, Romantic Movies
4                    #Roxy                          Comedies, Romantic Movies
20      100 Days My Prince  International TV Shows, Romantic TV Shows, TV ...
21    100 Days Of Solitude                Documentaries, International Movies
39              14 Cameras                           Horror Movies, Thrillers
...                    ...                                                ...
8776                   Zoo   Dramas, Independent Movies, International Movies
8795             يوم الدين   Dramas, Independent Movies, International Movies
8797          ​Goli Soda 2   Action & Adventure, Dramas, International Movies
8801  ​​Kuch Bheege Alfaaz   Dramas, Independent Movies, International Movies
8806   최강전사 미니특공대 : 영웅의 탄생                           Children & Family Movies

[1147 rows x 2 columns]

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

In [196]:
df_merged.loc[df_merged['director'].isnull(), ['title']]

title
8                                #blackAF
12                                    '89
14                               (Un)Well
20                     100 Days My Prince
21                   100 Days Of Solitude
...                                   ...
8796                  แผนร้ายนายเจ้าเล่ห์
8800  ​SAINT SEIYA: Knights of the Zodiac
8802                             忍者ハットリくん
8803                                 海的儿子
8804                                 마녀사냥

[2634 rows x 1 columns]

2. **Acceso a datos específicos con `iloc`:**

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

   

In [197]:
df_merged.iloc[:5, [8, 9, 11]]

title                     director      country
0                         #Alive                       Cho Il  South Korea
1  #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto        Italy
2              #FriendButMarried                Rako Prijanto    Indonesia
3            #FriendButMarried 2                Rako Prijanto    Indonesia
4                          #Roxy              Michael Kennedy       Canada

- Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.

    

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.

In [198]:
df_merged.iloc[-5:, [8, -1]] 

title                                        description
8802             忍者ハットリくん  Hailing from the mountains of Iga, Kanzo Hatto...
8803                 海的儿子  Two brothers start a new life in Singapore, wh...
8804                 마녀사냥  Four Korean celebrity men and guest stars of b...
8805              반드시 잡는다  After people in his town start turning up dead...
8806  최강전사 미니특공대 : 영웅의 탄생  Miniforce, a special task force of elite range...

- Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

In [199]:
df_merged.iloc[:10, 1:5]

Genre Premiere  Runtime  IMDB Score
0   NaN      NaN      NaN         NaN
1   NaN      NaN      NaN         NaN
2   NaN      NaN      NaN         NaN
3   NaN      NaN      NaN         NaN
4   NaN      NaN      NaN         NaN
5   NaN      NaN      NaN         NaN
6   NaN      NaN      NaN         NaN
7   NaN      NaN      NaN         NaN
8   NaN      NaN      NaN         NaN
9   NaN      NaN      NaN         NaN

- Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.




In [200]:
df_merged.iloc[-7:, [8, 9, 11]]

title       director      country
8800  ​SAINT SEIYA: Knights of the Zodiac            NaN        Japan
8801                 ​​Kuch Bheege Alfaaz           Onir        India
8802                             忍者ハットリくん            NaN        Japan
8803                                 海的儿子            NaN          NaN
8804                                 마녀사냥            NaN  South Korea
8805                              반드시 잡는다  Hong-seon Kim  South Korea
8806                  최강전사 미니특공대 : 영웅의 탄생  Young Jun Lee          NaN

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.


In [201]:
df_merged.iloc[14:25, :4]

Title Genre Premiere  Runtime
14   NaN   NaN      NaN      NaN
15   NaN   NaN      NaN      NaN
16   NaN   NaN      NaN      NaN
17   NaN   NaN      NaN      NaN
18   NaN   NaN      NaN      NaN
19   NaN   NaN      NaN      NaN
20   NaN   NaN      NaN      NaN
21   NaN   NaN      NaN      NaN
22   NaN   NaN      NaN      NaN
23   NaN   NaN      NaN      NaN
24   NaN   NaN      NaN      NaN

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posic

In [202]:
df_merged.iloc[20:30, [8, 13, 14]]

title  release_year rating
20                        100 Days My Prince          2018  TV-14
21                      100 Days Of Solitude          2018  TV-MA
22                                100 Humans          2020  TV-14
23                                100 Meters          2016  TV-MA
24       100 Things to do Before High School          2014   TV-Y
25  100 Years: One Woman's Fight for Justice          2016  TV-14
26                                100% Halal          2020  TV-14
27                               100% Hotter          2017  TV-14
28                           1000 Rupee Note          2014  TV-14
29                              12 ROUND GUN          2017  TV-MA

In [203]:
df_merged.to_csv ("datos/netflix_outer.csv")
df_merged

Title Genre Premiere  Runtime  IMDB Score Language show_id     type  \
0      NaN   NaN      NaN      NaN         NaN      NaN   s2037    Movie   
1      NaN   NaN      NaN      NaN         NaN      NaN   s2305    Movie   
2      NaN   NaN      NaN      NaN         NaN      NaN   s2482    Movie   
3      NaN   NaN      NaN      NaN         NaN      NaN   s2325    Movie   
4      NaN   NaN      NaN      NaN         NaN      NaN   s5974    Movie   
...    ...   ...      ...      ...         ...      ...     ...      ...   
8802   NaN   NaN      NaN      NaN         NaN      NaN   s6178  TV Show   
8803   NaN   NaN      NaN      NaN         NaN      NaN   s4915  TV Show   
8804   NaN   NaN      NaN      NaN         NaN      NaN   s7102  TV Show   
8805   NaN   NaN      NaN      NaN         NaN      NaN   s5023    Movie   
8806   NaN   NaN      NaN      NaN         NaN      NaN   s7109    Movie   

                              title                     director  \
0                            #Alive                       Cho Il   
1     #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto   
2                 #FriendButMarried                Rako Prijanto   
3               #FriendButMarried 2                Rako Prijanto   
4                             #Roxy              Michael Kennedy   
...                             ...                          ...   
8802                       忍者ハットリくん                          NaN   
8803                           海的儿子                          NaN   
8804                           마녀사냥                          NaN   
8805                        반드시 잡는다                Hong-seon Kim   
8806            최강전사 미니특공대 : 영웅의 탄생                Young Jun Lee   

                                                   cast      country  \
0                              Yoo Ah-in, Park Shin-hye  South Korea   
1                           Helen Mirren, Gengher Gatti        Italy   
2     Adipati Dolken, Vanesha Prescilla, Rendi Jhon,...    Indonesia   
3     Adipati Dolken, Mawar de Jongh, Sari Nila, Von...    Indonesia   
4     Jake Short, Sarah Fisher, Booboo Stewart, Dann...       Canada   
...                                                 ...          ...   
8802                                                NaN        Japan   
8803  Li Nanxing, Christopher Lee, Jesseca Liu, Appl...          NaN   
8804  Si-kyung Sung, Se-yoon Yoo, Dong-yup Shin, Ji-...  South Korea   
8805                                      Baek Yoon-sik  South Korea   
8806  Um Sang-hyun, Yang Jeong-hwa, Jeon Tae-yeol, S...          NaN   

              date_added  release_year rating   duration  \
0      September 8, 2020          2020  TV-MA        NaN   
1           July 1, 2020          2019  TV-14        NaN   
2           May 21, 2020          2018   TV-G        NaN   
3          June 28, 2020          2020   TV-G    104 min   
4         April 10, 2019          2018  TV-14        NaN   
...                  ...           ...    ...        ...   
8802   December 23, 2018          2012  TV-Y7  2 Seasons   
8803      April 27, 2018          2016  TV-14        NaN   
8804   February 19, 2018          2015  TV-MA        NaN   
8805   February 28, 2018          2017  TV-MA    110 min   
8806   September 1, 2018          2018  TV-Y7     68 min   

                                              listed_in  \
0        Horror Movies, International Movies, Thrillers   
1                   Documentaries, International Movies   
2         Dramas, International Movies, Romantic Movies   
3         Dramas, International Movies, Romantic Movies   
4                             Comedies, Romantic Movies   
...                                                 ...   
8802                             Anime Series, Kids' TV   
8803                  International TV Shows, TV Dramas   
8804  International TV Shows, Korean TV Shows, Stand...   
8805            Dramas, International Movies, Thrillers   
8806                           Children 